maven_iuvs/download.py

```diff
@@ -417,11 +417,12 @@ def sync_data(spice=True, l1b=True,
 
             print('Syncing ' + str(len(files_from_production)) +
                   ' files from production...')
-            call_rsync(login+production_l1b,
-                       l1b_dir,
-                       iuvs_vm_password,
-                       extra_flags=('--files-from=' +
-                                    transfer_from_production_file.name))
+            if len(prod_filenames) > 0:
```

Comment:
If we add this line here, we should probably add a similar line below at line 435, or (better) factor out this bit of code so that changes don't need to be made in more than one place.

 But actually my preference would be to just comment out the print statement on line 159, since running the rsync command doesn't hurt anything if no files are moved. I had this commented out originally and added it back in to do some debugging. Would that solve your issue Kei? If not, we can make the larger change to factor out the relevant code.

Reply:
You can continue to commit to this branch and the pull request will update automatically.

You said that you get an error when the code is called with no file? What error do you get?

Reply:
Ok, I'll check with this branch then. 
Below is the original error before I've added this pull request.

In [13]: maven_iuvs.download.sync_data(spice=False, l1b=True, minorb=12800, max
    ...: orb=12900)
input password for kema5665@maven-iuvs-itf: 
Fetching names of level 1B production and stage files from the VM...
                                                                               Syncing 0 files from production...
running rsync_command: rsync -trvzL --progress --files-from=/var/folders/ps/l1f7kmcx4997m2lywdkfnm7m0000gn/T/tmpf3px6esj kema5665@maven-iuvs-itf:/maven_iuvs/production/products/level1b/ /Volumes/Fenix/work/data/maven_iuvs/level1b/
---------------------------------------------------------------------------
EOF                                       Traceback (most recent call last)
~/opt/anaconda3/lib/python3.7/site-packages/pexpect/expect.py in expect_loop(self, timeout)
    110                 # Still have time left, so read more data
--> 111                 incoming = spawn.read_nonblocking(spawn.maxread, timeout)
    112                 if self.spawn.delayafterread is not None:

~/opt/anaconda3/lib/python3.7/site-packages/pexpect/pty_spawn.py in read_nonblocking(self, size, timeout)
    499         if (timeout != 0) and select(timeout):
--> 500             return super(spawn, self).read_nonblocking(size)
    501 

~/opt/anaconda3/lib/python3.7/site-packages/pexpect/spawnbase.py in read_nonblocking(self, size, timeout)
    175             self.flag_eof = True
--> 176             raise EOF('End Of File (EOF). Empty string style platform.')
    177 

EOF: End Of File (EOF). Empty string style platform.

During handling of the above exception, another exception occurred:

EOF                                       Traceback (most recent call last)
<ipython-input-13-65e421725022> in <module>()
----> 1 maven_iuvs.download.sync_data(spice=False, l1b=True, minorb=12800, maxorb=12900)

~/work/python_git/maven/lasp_iuvs/maven_iuvs/maven_iuvs/download.py in sync_data(spice, l1b, pattern, minorb, maxorb, include_cruise)
    422                        iuvs_vm_password,
    423                        extra_flags=('--files-from=' +
--> 424                                     transfer_from_production_file.name))
    425 
    426             # stage, identical to above

~/work/python_git/maven/lasp_iuvs/maven_iuvs/maven_iuvs/download.py in call_rsync(remote_path, local_path, ssh_password, extra_flags)
    163     cpl = child.compile_pattern_list(['.* password: ',
    164                                       '[0-9]+%'])
--> 165     child.expect_list(cpl)
    166 
    167     if 'password' in child.after:

~/opt/anaconda3/lib/python3.7/site-packages/pexpect/spawnbase.py in expect_list(self, pattern_list, timeout, searchwindowsize, async_, **kw)
    367             return expect_async(exp, timeout)
    368         else:
--> 369             return exp.expect_loop(timeout)
    370 
    371     def expect_exact(self, pattern_list, timeout=-1, searchwindowsize=-1,

~/opt/anaconda3/lib/python3.7/site-packages/pexpect/expect.py in expect_loop(self, timeout)
    115                     timeout = end_time - time.time()
    116         except EOF as e:
--> 117             return self.eof(e)
    118         except TIMEOUT as e:
    119             return self.timeout(e)

~/opt/anaconda3/lib/python3.7/site-packages/pexpect/expect.py in eof(self, err)
     61             if err is not None:
     62                 msg = str(err) + '\n' + msg
---> 63             raise EOF(msg)
     64 
     65     def timeout(self, err=None):

EOF: End Of File (EOF). Empty string style platform.
<pexpect.pty_spawn.spawn object at 0x1061c72d0>
command: /usr/bin/rsync
args: [b'/usr/bin/rsync', b'-trvzL', b'--progress', b'--files-from=/var/folders/ps/l1f7kmcx4997m2lywdkfnm7m0000gn/T/tmpf3px6esj', b'kema5665@maven-iuvs-itf:/maven_iuvs/production/products/level1b/', b'/Volumes/Fenix/work/data/maven_iuvs/level1b/']
buffer (last 100 chars): ''
before (last 100 chars): ' to consider\r\n\r\nsent 8 bytes  received 13 bytes  14.00 bytes/sec\r\ntotal size is 0  speedup is 0.00\r\n'
after: <class 'pexpect.exceptions.EOF'>
match: None
match_index: None
exitstatus: None
flag_eof: True
pid: 93602
child_fd: 23
closed: False
timeout: 30
delimiter: <class 'pexpect.exceptions.EOF'>
logfile: None
logfile_read: None
logfile_send: None
maxread: 2000
ignorecase: False
searchwindowsize: None
delaybeforesend: 0.05
delayafterclose: 0.1
delayafterterminate: 0.1
searcher: searcher_re:
    0: re.compile('.* password: ')
    1: re.compile('[0-9]+%')

Reply:
oh, ok. actually this is a problem we should solve in the call_rsync function by modifying the first pexpect.expect command. I'll try to push a commit and see if it fixes your problem.